lib/matplotlib/texmanager.py

```diff
@@ -103,33 +110,48 @@ def __new__(cls):
         Path(cls.texcache).mkdir(parents=True, exist_ok=True)
         return object.__new__(cls)
 
+    @_api.deprecated("3.6")
     def get_font_config(self):
+        preamble, font_cmd = self._get_font_preamble_and_command()
+        # Add a hash of the latex preamble to fontconfig so that the
+        # correct png is selected for strings rendered with same font and dpi
+        # even if the latex preamble changes within the session
+        preambles = preamble + font_cmd + self.get_custom_preamble()
+        return hashlib.md5(preambles.encode('utf-8')).hexdigest()
+
+    @classmethod
+    def _get_font_family_and_reduced(cls):
         ff = rcParams['font.family']
         ff_val = ff[0].lower() if len(ff) == 1 else None
         reduced_notation = False
-        if len(ff) == 1 and ff_val in self._font_families:
-            self._font_family = ff_val
-        elif len(ff) == 1 and ff_val in self._font_info:
-            reduced_notation = True
-            self._font_family = self._font_types[ff_val]
+        if len(ff) == 1 and ff_val in cls._font_families:
+            return ff_val, False
+        elif len(ff) == 1 and ff_val in cls._font_preambles:
+            return cls._font_types[ff_val], True
         else:
             _log.info('font.family must be one of (%s) when text.usetex is '
                       'True. serif will be used by default.',
-                      ', '.join(self._font_families))
-            self._font_family = 'serif'
-
-        fontconfig = [self._font_family]
-        fonts = {}
-        for font_family in self._font_families:
-            if reduced_notation and self._font_family == font_family:
-                fonts[font_family] = self._font_info[ff_val]
+                      ', '.join(cls._font_families))
+            return 'serif', False
+
+    @classmethod
+    def _get_font_preamble_and_command(cls):
+        requested_family, is_reduced_font = cls._get_font_family_and_reduced()
+
+        preambles = {}
+        for font_family in cls._font_families:
+            if is_reduced_font and font_family == requested_family:
+                preambles[font_family] = cls._font_preambles[
+                    rcParams['font.family'][0].lower()]
```

Comment:
Um, what is this? I thought `rcParams['font.family']` is a string?!? Should this be
```suggestion
                    rcParams['font.family'].lower()]
```

Reply:
No, it's a list of strings: https://github.com/matplotlib/matplotlib/blob/f670fe78795b18eb1118707721852209cd77ad51/lib/matplotlib/rcsetup.py#L914

Reply:
Ah, I was checking
https://github.com/matplotlib/matplotlib/blob/f670fe78795b18eb1118707721852209cd77ad51/lib/matplotlib/mpl-data/matplotlibrc#L219
and 
https://github.com/matplotlib/matplotlib/blob/f670fe78795b18eb1118707721852209cd77ad51/lib/matplotlib/mpl-data/matplotlibrc#L262

We should make that clearer in `matplotlibrc`.

And yes interestingly `validate_string_list` can make a list out of a single string. `matplotlibrc` is less well-defined than I thought 😦.

Reply:
This could reasonably be tracked as a separate issue?

Reply:
Sure: #22746.